### DISCUSSION TITLE
Should we remove flags that represent part of the API?

### DISCUSSION THREAD

**Discussion by keckler** [terrapower/armi#1833] [General]
There are currently flags for `Core` and `Reactor` being defined in our `flags` module:
https://github.com/terrapower/armi/blob/9dc6760de7da3800cbe98e83066cf5f343e81a6a/armi/reactor/flags.py#L208-L209

I would like to propose that we get rid of the two flags, as well as any other flags whose functionality is duplicated by other parts of the API (if any). My reasoning is the following:
1. These two flags do not serve any purpose that cannot also be fulfilled by something like `isinstance(myCoreObject, Core)`
2. [Flags can lead to unintended behavior](https://github.com/terrapower/armi/blob/9dc6760de7da3800cbe98e83066cf5f343e81a6a/armi/reactor/flags.py#L50-L66), expecially when we autoflag things ([which we do](https://github.com/terrapower/armi/blob/9dc6760de7da3800cbe98e83066cf5f343e81a6a/armi/reactor/flags.py#L22-L26))

A concrete example of this causing problems is documented in #1754 . Removing the `Core` and `Reactor` flags would be one way for us to safeguard against these types of problems, without us losing any actual functionality in the framework.

**Comment by john-science** [comment]
## Flags should define what something is

I have not seen this in the code base, and if it exists I want to squash it immediately:

https://github.com/terrapower/armi/blame/9dc6760de7da3800cbe98e83066cf5f343e81a6a/armi/reactor/flags.py#L64-L66

I really hope that is a problem of the past.


## CORE and REACTOR are bad flags

I agree with Chris above, that CORE and REACTOR are bad flags to exist. They are only used in a few places in ARMI:

https://github.com/terrapower/armi/blob/9dc6760de7da3800cbe98e83066cf5f343e81a6a/armi/reactor/reactors.py#L126
https://github.com/terrapower/armi/blob/9dc6760de7da3800cbe98e83066cf5f343e81a6a/armi/bookkeeping/db/database3.py#L860
https://github.com/terrapower/armi/blob/9dc6760de7da3800cbe98e83066cf5f343e81a6a/armi/bookkeeping/db/database3.py#L1422
https://github.com/terrapower/armi/blob/9dc6760de7da3800cbe98e83066cf5f343e81a6a/armi/reactor/reactors.py#L101

These are easily replaced with type checks. (Also, I cannot find usages of these two flags downstream of ARMI.)

## What other flags are unused or counter-useful?

I am immediately suspicious of these three flags, which are used nowhere but the docs:

https://github.com/terrapower/armi/blob/9dc6760de7da3800cbe98e83066cf5f343e81a6a/armi/reactor/flags.py#L204-L206

**Comment by opotowsky** [comment]
https://github.com/terrapower/armi/pull/1835 Closes this

**Reply by opotowsky** [reply]
oops @john-science was that correct? I see you listed 3 more unused flags
